auth_mellon_util.c

```diff
@@ -1507,32 +1507,104 @@ am_get_service_url(request_rec *r, LassoProfile *profile, char *service_name)
     return url;
 }
 
-/* Thus function checks if an HTTP PAOS header is valid.
+/* Removes leading & trailing whitespace and beginning & ending double quotes
  *
- * A PAOS header must be composed of 2 values seperated by a semicolon.
+ * Operates in place by modifying string.
  *
- * The first value must be a version declaration in the form ver="xxx"
- * (note the version string must be in double quotes).
+ * Returns pointer to modified string.
  *
- * The second value must be the ECP service.
- * (note the service string must be in double quotes).
+ * Example: " \"foo\"\n" -> "foo"
+ */
+
+char *
+am_strip_whitespace_and_quotes(char *str)
+{
+    char *begin, *end;
+
+    /* Skip leading whitespace */
+    for (begin = str; *begin && isspace(*begin); begin++);
+
+    /* Find end of string */
+    for (end = begin; *end; end++);
+
+    /* Skip trailing whitespace */
+    for (end = end > begin ? end-1 : end; end > begin && isspace(*end); end--);
```

Comment:
This logic is complicated and hard to read. I suggest extracting the test for `end == begin` out of this for loop and doing an early return in that case -- it is a special case where the string consists of only whitespace. E.g.:

```
if (begin == end) {
  return begin; /* The string consisted of only whitespace. */
}
```